eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  I. Non-static Logger instance
+ *  II. Logger variable name should be 'logger'
+ *  III. Guard with conditional block check whenever logger's logging method called.
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String LOGGER_CLASS = "Logger";
+    private static final String LOGGER_NAME = "logger";
+
+    private static final String LOGGER_NAME_ERR = "Incorrect name for Logger: use 'logger' instead of '%s' as Logger name";
+    private static final String LOGGER_LEVEL_ERR = "Please guard %s method call with %s method";
+    private static final String STATIC_LOGGER_ERR = "Logger should not be static: remove static modifier";
+    private static final int[] TOKENS = new int[] {
+        TokenTypes.METHOD_CALL,
+        TokenTypes.VARIABLE_DEF
+    };
+
+    private static final Map<String, String> loggerMethodMap = new HashMap<>();
+
+    // Logger level static final string variables
+    private static final String FATAL = "fatal";
+    private static final String ERROR = "error";
+    private static final String WARN = "warn";
+    private static final String INFO = "info";
+    private static final String DEBUG = "debug";
+    private static final String TRACE = "trace";
+
+    private static final String IS_FATAL_ENABLED = "isFatalEnabled";
+    private static final String IS_ERROR_ENABLED = "isErrorEnabled";
+    private static final String IS_WARN_ENABLED = "isWarnEnabled";
+    private static final String IS_INFO_ENABLED = "isInfoEnabled";
+    private static final String IS_DEBUG_ENABLED = "isDebugEnabled";
+    private static final String IS_TRACE_ENABLED = "isTraceEnabled";
+
+    private static final Set<String> loggerSet = new HashSet<>();
+
+    @Override
+    public void init() {
+        initLogLevelMap();
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return TOKENS;
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        // skip check if no Logger class imported
+        if (!hasLoggerImport(ast)) {
+            return;
+        }
+
+        switch (ast.getType()) {
+            case TokenTypes.VARIABLE_DEF:
+                isNameLogger(ast);
+                isNonStaticLoggerInstance(ast);
+                break;
+            case TokenTypes.METHOD_CALL:
+                isGuardConditionChecked(ast);
+                break;
+        }
+    }
+
+    /**
+     * Check if the file has Logger class imported
+     * @param rootAST The root of AST
+     * @return true if the class has Logger class imported.
+     */
+    private boolean hasLoggerImport(DetailAST rootAST) {
+        DetailAST importToken = rootAST.findFirstToken(TokenTypes.IMPORT);
```

Comment:
There's an incorrect assumption that the node is the "rootAST". The node passed in could be a variable_def or method_def from the usage in visitToken(). Consequently, you're walking up the tree multiple times to perform this check.

Reply:
that is correct. Will update the PR.